src/Extensions/ShareDraftContentFileShortcodeProviderExtension.php

```diff
@@ -20,16 +20,27 @@ public function updateGrant(bool &$grant, ?File $record = null, ?array $args = n
         if ($grant) {
             return;
         }
-        $session = $session = Controller::curr()->getRequest()->getSession();
+
+        $controller = Controller::curr();
+
+        if (!$controller) {
+            return;
+        }
+
+        $session = $controller->getRequest()->getSession();
```

Comment:
```suggestion
        if (!Controller::has_curr()) {
            return;
        }

        $session = Controller::curr()->getRequest()->getSession();
```
`Controller::curr()` throws an error if there's no controller. We have to check if there's a current controller with `has_curr()`

Reply:
Good to know! I wasn't aware of that.

Regarding the whitespace: Spaces before/after control structures is a Slevomat standard. I think it's quite a popular standard - or at least, we use it for all of our bespoke projects.